test/acl_usm_test.cpp

```diff
@@ -1208,4 +1246,78 @@ MT_TEST(acl_usm, memfill_usm) {
   ACL_LOCKED(acl_print_debug_msg("end memfill_usm\n"));
 }
 
-#endif // __arm__
+MT_TEST(acl_usm, read_device_global) {
+  ACL_LOCKED(acl_print_debug_msg("begin read_device_global\n"));
+  char str[100];
+  const size_t strsize = sizeof(str) / sizeof(char); // includes NUL (!)
+  char resultbuf[strsize];
+  cl_int status;
+  cl_event read_event = 0;
+
+  // Prepare host memory
+  syncThreads();
+  // Host pointer example
+  void *src_ptr = malloc(strsize);
+  CHECK(src_ptr != NULL);
+
+  syncThreads();
+
+  // Read from device global
+  status = clEnqueueReadGlobalVariableINTEL(m_cq, m_program, "dev_global_name",
+                                            CL_FALSE, strsize, 0, src_ptr, 0,
+                                            NULL, &read_event);
+  CHECK_EQUAL(CL_SUCCESS, status);
+
+  // Read from device global failed
+  status = clEnqueueReadGlobalVariableINTEL(
+      m_cq, m_program, "dev_global_name_not_exist", CL_FALSE, strsize, 0,
+      src_ptr, 0, NULL, &read_event);
+  CHECK_EQUAL(CL_INVALID_ARG_VALUE, status);
+
+  // Block on all event completion
+  CHECK_EQUAL(CL_SUCCESS, clFinish(m_cq));
+
+  // Host pointer example
+  free(src_ptr);
+
+  ACL_LOCKED(acl_print_debug_msg("end read_device_global\n"));
+}
+
+MT_TEST(acl_usm, write_device_global) {
+  ACL_LOCKED(acl_print_debug_msg("begin write_device_global\n"));
+  char str[100];
+  const size_t strsize = sizeof(str) / sizeof(char); // includes NUL (!)
+  char resultbuf[strsize];
+  cl_int status;
+  cl_event write_event = 0;
+
+  // Prepare host memory
+  syncThreads();
+  // Host pointer example
+  void *src_ptr = malloc(strsize);
+  CHECK(src_ptr != NULL);
+
+  syncThreads();
+
+  // Write device global
+  status = clEnqueueReadGlobalVariableINTEL(m_cq, m_program, "dev_global_name",
```

Comment:
Looks like this should be clEnqueueWriteGlobalVariableINTEL

Reply:
Fixed